src/index/CompressedRelation.h

```diff
@@ -183,16 +185,6 @@ class CompressedRelationWriter {
     outfile_.close();
   }
 
-  /// Get the complete CompressedRelationMetaData created by the calls to
-  /// addRelation. This meta data is then deleted from the
-  /// CompressedRelationWriter. The typical workflow is: add all relations,
-  /// then call `finish()` and then call this method.
-  auto getFinishedMetaData() {
-    auto result = std::move(metaDataBuffer_);
-    metaDataBuffer_.clear();
-    return result;
-  }
-
```

Comment:
Can you do something similar for the `getFinishedBlocks()`
We currently have to manually call `finish` and then call `getFinishedBlocks()`.
It would be better to call `finish` inside the `getFinishedBlocks()`,
make `finish` private, and force the `Writer` to be moved in order to call `getFinishedBlocks()`:
````
vector<BlockData> getFinishedBlocks()&& {
   finish();
  return std::move(blocks);
}
```
This makes the calling code much simpler as we don't have to remember to `finish`.